src/components/SlateEditor/plugins/video/EditVideo.tsx

```diff
@@ -74,46 +82,78 @@ export const brightcoveEmbedFormRules: RulesType<FormValues> = {
   },
 };
 
-const EditVideo = ({ embed, saveEmbedUpdates, activeSrc, close, setHasError }: Props) => {
+const EditVideo = ({ embed, editor, element, setHasError }: Props) => {
   const { t } = useTranslation();
 
-  const initialValues = useMemo(() => toVideoEmbedFormValues(embed), [embed]);
+  const [open, setOpen] = useState(false);
+
+  const onClose = () => {
+    setOpen(false);
+    ReactEditor.focus(editor);
+    const path = ReactEditor.findPath(editor, element);
+    if (Editor.hasPath(editor, Path.next(path))) {
+      setTimeout(() => {
+        Transforms.select(editor, Path.next(path));
+      }, 0);
+    }
+  };
+
+  const activeSrc = (embed: BrightcoveMetaData) => {
+    const { account, videoid } = embed.embedData;
+    const startTime = getBrightCoveStartTime(videoid);
+    const id = addBrightCoveTimeStampVideoid(videoid, startTime);
+    return `https://players.brightcove.net/${account}/${config.brightcoveEdPlayerId}_default/index.html?videoId=${id}`;
+  };
 
-  const handleSave = (values: FormValues) => {
-    saveEmbedUpdates({
-      alt: values.alttext,
-      caption: inlineContentToHTML(values.caption),
-      videoid: addBrightCoveTimeStampVideoid(embed.videoid, values.startTime),
-    });
-    close();
+  const initialValues = useMemo(() => toVideoEmbedFormValues(embed?.embedData), [embed]);
+
+  const onSave = (values: FormValues) => {
+    Transforms.setNodes(
+      editor,
+      {
+        data: {
+          ...embed.embedData,
+          alt: values.alttext,
+          caption: inlineContentToHTML(values.caption),
+          videoid: addBrightCoveTimeStampVideoid(embed?.embedData?.videoid, values.startTime),
+        },
+      },
+      {
+        match: (node) => Element.isElement(node) && node.type === TYPE_EMBED_BRIGHTCOVE,
+        at: ReactEditor.findPath(editor, element),
+      },
+    );
+    onClose();
   };
 
   return (
-    <>
-      <ModalHeader>
-        <ModalTitle>{t("form.video.editVideo")}</ModalTitle>
-        <ModalCloseButton />
-      </ModalHeader>
-      <ModalBody>
-        <SlateVideoWrapper>
-          <StyledVideo
-            title={`Video: ${embed.metaData ? embed.metaData.name : ""}`}
-            frameBorder="0"
-            src={activeSrc}
-            allowFullScreen
-          />
-        </SlateVideoWrapper>
-        <Formik
-          initialValues={initialValues}
-          validate={(values) => validateFormik(values, brightcoveEmbedFormRules, t)}
-          validateOnBlur={false}
-          validateOnMount
-          onSubmit={handleSave}
-        >
-          {(formik) => <VideoEmbedForm {...formik} setHasError={setHasError} close={close} />}
-        </Formik>
-      </ModalBody>
-    </>
+    <Modal open={open} onOpenChange={setOpen}>
```

Comment:
Gir det mer mening å styre dette fra SlateVideo?

Reply:
Altså at Modal, ModalTrigger og Content brukes der

Reply:
Følte det virka ryddigere å ha de i `EditVideo` når konteksten om at modalen er oppe ikke påvirker noe i `SlateVideo` . Om du vil heller ønsker at så kan jeg godt slenge det inn i SlateVideo istedenfor, bare gi meg et pip

Reply:
Tja, vet ikke jeg. Tenkte det var ryddig å sørge for at EditVideo kunne brukes andre steder, men sikkert ikke så viktig